server/scrape-negative

```diff
@@ -3,28 +3,79 @@
 from helpers.validator import validate_venv
 validate_venv()
 
-import sys, argparse
+import sys, argparse, random
 
 from definitions import NEGATIVE_QUERIES
 from scraper.scrape_news import scrape_news_pages
 from nlp.nlp import NLProcessor
+from database.mock_database import DatabaseNegative
+
+
+def sample_fitness(sample):
+    #fitness of a sample is the mean pairwise distance between all summaries in the sample,
+    #not double-counting the distances
+    fitness = 0
+    while len(sample) > 0:
+        base = sample[0]
+        sample = sample[1:]
+        for element in sample:
+            fitness += NLProcessor.pairdistance(base, element)
+    return fitness
+
+def sample_randomsearch(dataset, samplesize, iterations):
+    previous_fitness = 0
+    previous_sample = list()
+    for i in range(iterations):
+        sample = random.sample(dataset, samplesize)
+        fitness = sample_fitness(sample)
+        if fitness > previous_fitness:
+            previous_sample = sample
+            previous_fitness = fitness
+    return previous_sample
+
+def find_best_match(cluster_keys, summary):
+    cluster_key_similarities = [NLProcessor.pairdistance(key, summary) for key in cluster_keys]
+    return cluster_keys[cluster_key_similarities.index(min(cluster_key_similarities))]
+
+
+def create_clusters(samplesize, iterations):
+    dataset = DatabaseNegative.get_all_summaries()
+    cluster_keys = sample_randomsearch(dataset, samplesize, iterations)
+    clusters = dict()
+    for summary in dataset:
+        match = find_best_match(cluster_keys, summary)
+        if match in clusters.keys():
+            clusters[match].append(summary)
+        else:
+            clusters[match] = [summary]
+    DatabaseNegative.insert_new_clusters(clusters)
+
 
 if __name__ == "__main__":
     parser = argparse.ArgumentParser(prog="scraper")
     parser.add_argument("-n", "--narticles", type=int, action="store", default=10)
+    parser.add_argument("-c", "--cluster", action="store_true")
+    parser.add_argument("-e", "--exclusive_clustering", action="store_true")
```

Comment:
are these args mutually exclusive? if so, argparse can handle this with mutually exclusive groups